File: typescript/Behavioral/13-ChainOfResponsibility/Handler/ManagerApprover.ts
import { BaseApprovalHandler } from "./BaseApprovalHandler";
import { ApprovalRequest } from "../Model/ApprovalRequest";

export class ManagerApprover extends BaseApprovalHandler {
  private static readonly LIMIT = 20_000;

  handle(request: ApprovalRequest): void {
    if (request.cost <= ManagerApprover.LIMIT)
      console.log(`  ✅ Manager approved: ${request}`);
    else {
      console.log(
        `  ↗️  Manager escalating (N$${request.cost.toFixed(2)} > N$${ManagerApprover.LIMIT.toFixed(2)})...`,
      );
      super.handle(request);
    }
  }
}

File: csharp/Behavioral/13-ChainOfResponsibility/Handler/ManagerApprover.cs
public class ManagerApprover : BaseApprovalHandler
{
    private const decimal Limit = 20_000m;

    public override void Handle(ApprovalRequest request)
    {
        if (request.Cost <= Limit)
            Console.WriteLine($"  ✅ Manager approved: {request}");
        else
        {
            Console.WriteLine($"  ↗️  Manager escalating (N${request.Cost:N2} > N${Limit:N2})...");
            base.Handle(request);
        }
    }
}
